src/login/logind.c

```diff
@@ -986,6 +986,29 @@ static int manager_dispatch_idle_action(sd_event_source *s, uint64_t t, void *us
         return 0;
 }
 
+static int manager_parse_config_file(Manager *m) {
+        assert(m);
+
+        return config_parse_many(PKGSYSCONFDIR "/logind.conf",
+                                 CONF_PATHS_NULSTR("systemd/logind.conf.d"),
+                                 "Login\0",
+                                 config_item_perf_lookup, logind_gperf_lookup,
+                                 false, m);
+}
+
+static int manager_dispatch_reload_signal(sd_event_source *s, const struct signalfd_siginfo *si, void *userdata) {
+        Manager *m = userdata;
+        int r;
+
+        r = manager_parse_config_file(m);
+        if (r < 0)
+                log_warning_errno(r, "Failed to parse config file: %m");
+        else
+                log_info("Config file reloaded.");
+
+        return 0;
+}
```

Comment:
Not sure how I feel about this. After all this only adds in what's set in logind.conf now, but does not reset what it not set there. Which means, if you have a setting in logind.conf, then comment it and reload logind you won't get the setting reset, and will get a different result than you'd get if instead you had restarted logind.

THis is particularly problematic as KillOnlyUsers= and similar settings take lists of things, and on each SIGHUP we'd keep adding to this list...

To fix this properly, we'd need a manager_reset_config() call which resets all settings to the defaults, and is invoked before anager_parse_config_file() here...